app/src/main/java/de/westnordost/streetcomplete/view/LaurelWreath.kt

```diff
@@ -0,0 +1,94 @@
+package de.westnordost.streetcomplete.view
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.ColorFilter
+import android.graphics.Paint
+import android.graphics.PixelFormat
+import android.graphics.RectF
+import android.graphics.drawable.Drawable
+import androidx.core.graphics.withRotation
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.ktx.flipHorizontally
+import de.westnordost.streetcomplete.util.ktx.getBitmapDrawable
+
+/*
+Drawable providing decoration, suitable for a circular background
+100 and more: fully grown wreath with all pretty elements
+99 to 10: may be losing elements as it gets smaller
+below: no decorative styling at all
+ */
+class LaurelWreath(val resources: Resources, private val percentageOfGrowth: Int) : Drawable() {
+    val laurelLeafOnStalk = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_rotated)
+    val laurelStalk = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_stalk)
+    val horizontalEndingLeaf = resources.getBitmapDrawable(R.drawable.ic_laurel_leaf_ending)
+
+    private val antiAliasPaint: Paint = Paint().apply {
+        isAntiAlias = true
+        isFilterBitmap = true
+    }
+
+    override fun draw(canvas: Canvas) {
+        // Get the drawable's bounds
+        val canvasWidth: Int = bounds.width()
+        val canvasHeight: Int = bounds.height()
+        val circleRadius: Float = Math.min(canvasWidth, canvasHeight).toFloat() / 2f
+
+        val niceSubtleGreen: Paint = Paint().apply { setARGB(255, 152, 184, 126) }
+        canvas.drawCircle((canvasWidth / 2).toFloat(), (canvasHeight / 2).toFloat(), circleRadius, niceSubtleGreen)
+
+        if (percentageOfGrowth < 10) {
+            return
+        }
+
+        val decorationSegmentImageWidth = laurelLeafOnStalk.intrinsicWidth // width is the same as intrinsicWidth
+
+        val regularSegmentCount = 11f
+
+        val circleCenterX = canvasWidth / 2f
+        val shownSegments = ((regularSegmentCount - 1) * percentageOfGrowth / 100).toInt()
+        val locationBetweenCenterAndEdge = 0.78f
+        for (i in 1..shownSegments) {
+            // https://developer.android.com/reference/kotlin/androidx/core/graphics/package-summary#(android.graphics.Canvas).withRotation(kotlin.Float,kotlin.Float,kotlin.Float,kotlin.Function1)
+
+            var bitmap = laurelLeafOnStalk.bitmap
+            if (i == shownSegments) {
+                bitmap = horizontalEndingLeaf.bitmap
+            }
+            val stalkPaint = Paint()
+            stalkPaint.isAntiAlias = true
+            stalkPaint.setARGB(255, 0, 106, 0)
+            stalkPaint.style = Paint.Style.STROKE
+            stalkPaint.strokeWidth = 3f
```

Comment:
should be a private field. Doesn't Android Studio warn about new object allocations in the draw method?